src/custom-components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import { usePagination, DOTS } from "./usePagination";
+
+export interface PaginationInterface {
+  totalCount: number;
+  pageSize: number;
+  currentPage: number;
+  onPageChange: (arg: number) => void;
+}
+
+const Pagination: React.FC<PaginationInterface> = ({
+  totalCount,
+  pageSize,
+  currentPage,
+  onPageChange,
+}) => {
+  const { paginationRange } = usePagination({
+    totalCount,
+    pageSize,
+    currentPage,
+  });
```

Comment:
Oh, regarding this comment: https://github.com/FromDoppler/doppler-sms-webapp/pull/4#discussion_r1061626519

`paginationRange` is a value resulting from a pure function with inputs `totalCount`, `pageSize`, and `currentPage`. 

So, you do not need a hook, it is enough with a simple function, and it will be also easy to be tested.

Or am I missing something?

Reply:
It sounds good and it might work, I'm going to try it. even in that case, according to the [hooks reference](https://reactjs.org/docs/hooks-reference.html#usememo) it's still better to put it in useMemo for optimization.

..."Write your code so that it still works without useMemo — and then add it to optimize performance."...

Reply:
> [...] even in that case, according to the [hooks reference](https://reactjs.org/docs/hooks-reference.html#usememo) it's still better to put it in useMemo for optimization.
> 
> ..."Write your code so that it still works without useMemo — and then add it to optimize performance."...

Not necessarily, it depends on the expansiveness of the calculation and the usage. Take into account [what Donal Knuth said](https://stackify.com/premature-optimization-evil/)

Reply:
Well, in this case it could work, but we could have unnecessary executions for each render by not having access to an array of dependencies. But I'm going to try it and see what I find. In the end, the hooks give you certain guarantees so you don't have to worry about them.

Reply:
To close this thread. After doing the tests, it is confirmed that the functions are executed an unnecessary number of times in a single function (without the use of a hook like useMemo or useEffect). But talking to Andres he confirms that this doesn't matter, so we agreed to go ahead with a simple function.

Reply:
👍

> he confirms that this doesn't matter

Right. The function is executed more times, but the component is rendered fewer times because we avoid the `useEffect`.